### ISSUE TITLE
Connect to SQL DB from IIS

### ISSUE THREAD

**Issue by Adamster1981** [ProjectNami/projectnami#160]
Hello, I have created a Lab environments, no firewall or anti-virus in between, where IIS is running on one server and SQL server is running on another server.  I have used this step by step to set up projectname on the IIS server and also created database and its iuser.
http://support.projectnami.org/viewtopic.php?f=4&t=79

I am at step 27 and 28, where I am trying to connect to the database, I have temporarily ensure that full permission is there on the wwwroot folder to everyone (just so i can install).  I have also tried odbc connection from IIS to SQL server using the same username and database setting, however, wordpress installation keeps throwing an error saying establishing a database connection.  Any idea?

**Comment by patrickebates** [comment]
By default, many new SQL Server installations have TCP/IP disabled at the server.  I would check the protocols in the SQL Server Configuration Manager (they would be under SQL Server Network Configuration).  Any changes will require that you restart the service.

**Comment by Adamster1981** [comment]
Thank you I checked it and it is enable

**Comment by calway** [comment]
Am also trying to create a local Project Nami wordpress install to do some migration testing and have a lot of issues getting a VM working. Created a W2012R2 VM with IIS. Installed .NET3.5 and .NET4.6 and IIS and then php5.6.30 using Platform installer. I could not get the ODBC driver to install using platform installer so had to resort to downloading the SQLDRV32.EXE and simple copying the dll into the ext folder of php56. Als changed some stuff in php.ini And finally got wordpress installer to start. 
However I cannot get passed the setup-config page where you enter the sql server info. It keeps saying "Error establishing a database connection" and no wp-config.php is generated. I know I entered the correct information, and I tried to connect an ODBC with that info from this vm and that worked. 
P.S. the sql server is on another machine, it's sqlexpress 2016 as a default instance with tcp/ip enabled.
I read somewhere that the sql driver by default used windows authentication :-| but not sure if that is correct (the interwebs are full of bad info :-) )

Not sure whether Adamster1981 solved his problem and for me this VM is just a low prio test. We are currently running hosted but since we already have an SQL VM running it would be easy to simply setup a small windows VM to run Project Nami. We run outside of Azure by the way! Although I previously also tested a full azure install, and that was a breeze (except for moving the existing site data and plugins over, that was a lot of manual steps)

**Comment by patrickebates** [comment]
Have you tried installing PHP 7 and the matching SQL driver using WebPI?  Or is there a specific reason you need PHP 5.6?

**Comment by calway** [comment]
Installing now php7 now...... By the way I did not install de SQL driver using platform installer, that continualy did not install. So at last I resorted to simply putting the SQLSRV30.exe (5.6 dll in the extensions folder, That stopped Project Nami from detecting that the driver was not installed, but perhaps it is still missing some components (hmmm that might be it) but what parts...

**Comment by calway** [comment]
And inspecting the error log from platform installer I noticed it tried to install C:\Users\Administrator\AppData\Local\Microsoft\Web Platform Installer\installers\MSODBC13\B6A067E6AF8895D48B170643A49F2795FE2E7D0F\msodbcsql.msi which seemed not to work and I just ran that statement frm a cmd windows and it installed OK.
Revisiting the wordpress installer I did get passed the server not found and now it says it cannot save the wp-config.php, which is probably a permissions error.,
Wow these installations are very picky and problematic. Not what I am used with these types of install. Another reason to just host in Azure :-)

**Comment by patrickebates** [comment]
Welcome to why PN exists in the first place.  Ease of Azure Web Apps, ease of Azure SQL DB.

**Comment by Adamster1981** [comment]
FYI, I never got my issue resolved but I am still interested in resolving it.  Calway, you seem to have done a lot of troubleshooting and copying SQLSRV32 was a good idea, but I still would insist that we get this to work out of the box with SQL and Clustering.

**Comment by calway** [comment]
Tried to do a reinstall and hit my head several times again :-)

I am running IIS and SQL on separate machines. We use SQL Server 2016 Express where I created an empty database with it's own login to be used by wordpress (Project Nami)

I created a clean Windows 2012R2 VM on my PC and used server manager to install IIS
Don't forget to include ASP.NET4.5 and CGI under role services.

Then download Platform installer 5 and install. Search for PHP. I installed "PHP 5.6.30" since I need that version due to an older revolution slider plugin that we use :-(
I removed the "PHP Manager for IIS" and the "Microsoft Drivers 3.2 for PHP v5.6 for SQL Server in IIS" that get added automatically. Both don't install out of the box anyway.

Then Bing for "Download SQLSRV32" and use the first link: http://www.microsoft.com/en-us/download/details.aspx?id=20098
Download the SQLSRV32.EXE and extract it's contents. 
Then copy the php_sqlsrv_56_nts.dll and php_pdo_sqlsrv_56_nts.dll to the extensions folder of PHP
and edit php.ini to include this extension (scroll all the way down to the bottom and add a line: 

extension=php_sqlsrv_56_nts.dll

Then also download the correct VC runtime for PHP5.6 which is this one: http://www.microsoft.com/en-us/download/details.aspx?id=30679 Found this on this site: http://serverfault.com/questions/613927/500-error-on-fresh-install-of-iis-and-php
It seems that Platform installer does not include that, and it requires an older version

At first i proceeded to the next steps which gave connection problems which were resolved after an IIS restart. So do that now.

The last step is creating an IIS website and copy Project Nami into the root. Make sure the IUSR user had enough rights on the root folder (I gave it full control :-O )

Then I could succesfully go through the wordpress setup process.
